src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,349 @@
+package scala                     //TODO: Change this strawman --> we just need sizeHintIfCheap from scala
```

Comment:
Another question - I understand why knownSize might return -1 for say Streams or Iterators but what about somethlng like Lists? The idea in the previous API was the sizeHintIfCheap would only give you size in O(1) if possible. But, now, knownSize does not apply to List - for lists the size is known but is expensive O(n) to calculate. I think sizeIfCheap was more apt in that case.